_posts/2023-10-13-authz-keto-introduction.md

```diff
@@ -0,0 +1,273 @@
+
+* **Simplicity in Management**: The system was built on the hypothesis that no individual or team would dedicate their full time to configuring an authorization system.
+* **Intuitive Nature**: Keto is designed to be self-explanatory, minimizing the learning curve and ensuring that users find the system approachable.
+* **Familiarity**: By adhering to familiar paradigms and practices, Keto reduces the onboarding time for developers and system administrators.
+* **Ease of Modification**: Any changes or updates to the authorization configurations can be carried out with high confidence, ensuring system stability.
+* **Editorial Support**: Thanks to the Ory Permission Language, which is a subset of Typescript, Keto is compatible with numerous editors. This ensures users have the flexibility and familiarity in their choice of tools.
+
+## Core Concepts of Ory Keto
+
+1. **Adoption of the Relation Tuple Model**: Ory Keto adapts the Zanzibar-inspired relation tuples model. As we’ve previously elaborated, a relation tuple succinctly describes a relationship in a format like `namespace:object#relation@subject`. Through these tuples, Ory Keto can express and evaluate intricate hierarchical relationships and permissions across a multitude of objects and users, ensuring a granular and fine-tuned access control mechanism.
+2. **Ory Permission Language (OPL)**: To complement its adoption of relation tuples, Ory Keto introduces the Ory Permission Language (OPL). Conceptualized as a subset of TypeScript, OPL facilitates the definition and evaluation of policies and permissions within the Keto environment. By leveraging a familiar syntax style and structure, OPL provides developers with a more intuitive and streamlined process to articulate complex authorization policies, while maintaining efficiency and precision.
+
+### Ory Keto Namespaces
+
+In Ory Keto, namespaces are essentially the building blocks that aid in grouping related objects. To define these namespaces, Ory Keto uses TypeScript's class structures. 
+
+```
+import { Namespace, Context } from "@ory/keto-namespace-types"
+
+class User implements Namespace {}
+class Document implements Namespace {}
+class Folder implements Namespace {}
+```
+
+### Ory Keto Relations
+
+In Ory Keto, defining relations is at the heart of the model. Let's illustrate this with a hands-on example featuring two namespaces: Document and User.
+Within the Document namespace, a user is designated as either an owner, a viewer, or an editor of a document through specific relations.
+
+The core of authorization revolves around permissions, such as 'view' or 'edit', rather than checking relations like 'owner' or 'editor'. The concrete permission is checked against the relationships. Within the Ory Permission Language, permissions are declared as functions inside the permits property of the corresponding namespace. Let's explore how permissions might be outlined for our Document namespace:
+
+```
```

Comment:
formatting: use triple back tick with `ts` suffix to enable syntax highlighting. 

example:
```md
    ```ts
       class User {}
```

Reply:
assuming this language is a subset of typescript